modules/pipeline/providers/reconciler/pipeline_reconciler.go

```diff
@@ -89,12 +87,25 @@ type defaultPipelineReconciler struct {
 	// canceling
 	flagCanceling bool
 
-	// flag have task
-	flagHaveTask *bool
+	// task related
+	totalTaskNumber            *int
```

Comment:
maybe it's unnecessary to set `totalTaskNumber` pointer type, there are not found `totalTaskNumber` compare with nil

Reply:
pointer type is more clear to distinguish from truly zero task.